### PR TITLE
[HPARK-294] - Add additional columns

### PR DESCRIPTION
**Pull request by evaldez-gs**
**Changes:**

- Add 3 new columns: message_id (uuid), source (string) and campaign_user_id (uuid)
- Change main migration to consider last changes
- Change `short_url`, `generate` and `generate!` methods to support new optional columns
- Add changes to README.md (includes an additional migration to work)
- Add unit tests
- Update rspec version to run specs
- Update gem version to highlight new changes

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: README.rdoc [hunk 1]
```diff
@@ -206,6 +206,27 @@ Bonus: add an index if you plan on querying this +:meta+ column
...
+
+Data type +:uuid+ is not supported natively for sqlite
+
+  bundle exec rails g migration add_additional_columns_to_shortened_url
+
+
+  class AddAdditionalColumnsToShortenedUrl < ActiveRecord::Migration
+    def change
+      add_column :shortened_urls, :message_id, :uuid, index: true
+      add_column :shortened_urls, :source, :string, limit: 50
```
Comment (mchaisse) [inline README.rdoc]: Looking at the `migration.rb` file, `source` has an index, which is missing here.

File: lib/shortener/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module Shortener
-  VERSION = '1.0.2'
+  VERSION = '1.0.3'
```
Comment (mchaisse) [inline lib/shortener/version.rb]: Don't tag yet, this will be done during bump version process after merge.

**Review by mchaisse — CHANGES_REQUESTED** [review]
+ fix tests + what we talked about 😉

**Comment by evaldez-gs** [comment]
New changes:

- Change naming to be more generic for new columns
- Reuse owner column
- Change engine to postgresql to use `uuid` and `hstore`
- Change unit tests
- Change back to original gem version
- Update documentation
- Remove rvm versions < 2.2.0
- Comment one unit test because there is an issue with a postgresql transaction

**Comment by evaldez-gs** [comment]
@mchaisse I've changed it according to new changes listed in previous comment and last discussion about `travis`. Could you review it? Thanks!

File: README.rdoc [hunk 3]
```diff
@@ -21,6 +21,11 @@ Shortener is designed to work from within a Rails 3 and Rails 4 applications. It
 As Ruby 1.9.3 entered end of maintainance in early 2015, the last version of the Shortener gem
 to support Ruby 1.9.3 is 0.4.x. Shortener v0.5 onwards will require Ruby 2+.
 
+UPDATE:
+
+RVM versions +2.0.0+ and +2.1.0+ are not longer supported.
+Tested RVM versions: +2.2.3+ and +2.3.1+
```
Comment (mchaisse) [inline README.rdoc:27]: This section is part of `Ruby Version Support`

File: app/helpers/shortener/shortener_helper.rb [hunk 4]
```diff
@@ -3,13 +3,16 @@ module Shortener::ShortenerHelper
...
+                related_id: nil, source_type: nil)
     short_url = Shortener::ShortenedUrl.generate(url,
                                                  owner:      owner,
                                                  custom_key: custom_key,
                                                  expires_at: expires_at,
                                                  fresh:      fresh,
-                                                 meta:       meta)
+                                                 meta:       meta,
+                                                 related_id: related_id,
+                                                 source_type:     source_type)
```
Comment (mchaisse) [inline app/helpers/shortener/shortener_helper.rb]: too many spaces or align with the other parameters

File: spec/dummy/config/database.yml [hunk 5]
```diff
@@ -1,22 +1,21 @@
-# SQLite version 3.x
-#   gem install sqlite3
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
```
Comment (mchaisse) [inline spec/dummy/config/database.yml]: Prefer single quotes `'` if no interpolation.

File: spec/dummy/config/database.yml [hunk 6]
```diff
@@ -1,22 +1,21 @@
...
-# Warning: The database defined as "test" will be erased and
-# re-generated from your development database when you run "rake".
-# Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
-  database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
+  <<: *default
+  database: <%= ENV.fetch("DB_NAME") { 'shortener_test' } %>
```
Comment (mchaisse) [inline spec/dummy/config/database.yml]: same

File: README.rdoc (hunk 3 above)
Comment (evaldez-gs) [reply README.rdoc:27]: The comment is inside that section :)

**Comment by evaldez-gs** [comment]
I've fixed last changes as requested, just I didn't know what to change on README.md.

File: README.rdoc (hunk 3 above)
Comment (mchaisse) [reply README.rdoc:27]: 👍

**Review by mchaisse — APPROVED** [review]
LGTM

**Pull request merged**